src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -305,6 +305,72 @@ private static void FlattenDeconstructionMethods(DeconstructionInfo deconstructi
             }
         }
 
+        public IMethodSymbol GetElementAccessMethod(SemanticModel semanticModel, SyntaxNode node)
+        {
+            if (node is ElementAccessExpressionSyntax elementAccess)
+            {
+                var info = semanticModel.GetSymbolInfo(elementAccess);
+                var symbol = info.Symbol;
+
+                if (symbol != null && symbol.IsKind(SymbolKind.Method))
+                {
+                    return (IMethodSymbol)symbol;
+                }
+            }
+
+            return null;
+        }
+
+        public IPropertySymbol GetElementAccessCountableProperty(SemanticModel semanticModel, SyntaxNode node)
+        {
+            // This method returns the countable property associated with a given indexer, if one exists.
+            // We prioritize Length over Count if both are present in the indexer definition.
+            // See https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#implicit-index-support.
+            if (node is ElementAccessExpressionSyntax elementAccess)
+            {
+                var info = semanticModel.GetSymbolInfo(elementAccess);
+                var symbol = info.Symbol;
+
+                // If we can't find a valid symbol, compare against the first candidate symbol instead.
+                // TO-DO: not sure if we should be doing this or if there's a bug somewhere? Including this check makes certain tests pass.
+                if (symbol == null)
+                {
+                    symbol = info.CandidateSymbols.FirstOrDefault();
+                }
+
+                if (symbol == null || symbol.ContainingType == null)
+                {
+                    return null;
+                }
+
+                IPropertySymbol useCountPropertySymbol = null;
+                var containingType = symbol.ContainingType;
+                foreach (var location in containingType.Locations)
+                {
+                    var propertySymbols = semanticModel.LookupSymbols(location.SourceSpan.Start, containingType)
+                        .Where(s => s.IsKind(SymbolKind.Property));
```

Comment:
`containingType.GetMembers().OfType<IPropertySymbol>();`

(def remove the LookupSymbols, you sholud not have to do that at all).

Reply:
also, you can go GetMembers(LengthPropertyName) and GetMembers(CountPropertyName) for better efficiency

Reply:
also, the LookupSymbol is totally unsafe because you are comparing random semantic model with random location.